engine/player/unique_gear_thewarwithin.cpp

```diff
@@ -6919,6 +6919,239 @@ void zees_thug_hotline( special_effect_t& effect )
   new dbc_proc_callback_t( effect.player, effect );
 }
 
+// 443533 Equip Driver & Values
+// 443535 On Use
+// 443534 On Use Buff, radiance, double all effects
+// 450696 50 verses of Resillience tracking buff
+// 450699 50 verses of Radiance tracking buff
+// 450706 Inner Resillience Proc driver
+// 450720 Inner Radiance Proc driver
+// 450719 Magic Absorb buff
+// 450721 Ire of Devotion damage spell
+
+// 443533 Values
+// s1 - static value of 50
+// s2 - crit strike amount increase - Used in 450720 buff
+// s3 - devotion of ire damage - Used in 450721 spell
+// s4 - armor increase - Used in 450706 buff
+// s5 - magic absorb value - Used in 450719 buff
+void tome_of_lights_devotion( special_effect_t& effect )
+{
+  auto equip_driver = effect.player->find_spell( 443533 );
+  assert( equip_driver && "Tome of Lights Devotion equip driver not found" );
+
+  // Setup double value buff
+  auto radiance_buff = create_buff<buff_t>( effect.player, effect.player->find_spell( 443534 ) );
+
+  struct inner_resilience_cb_t : public dbc_proc_callback_t
+  {
+    struct ward_of_devotion_buff_t : public absorb_buff_t
+    {
+      ward_of_devotion_buff_t( const special_effect_t& e, const spell_data_t* s, const spell_data_t* data )
+        : absorb_buff_t( e.player, "ward_of_devotion", s )
+      {
+        cumulative = false;
+        set_default_value( data->effectN( 5 ).average( e ) );
+      }
+    };
+    buff_t* ward_of_devotion_buff;
+
+    const spell_data_t* data;
+
+    inner_resilience_cb_t( const special_effect_t& e )
```

Comment:
Probably dont need a custom callback for this if all it does is trigger a buff, can use `effect.custom_buff = buff_t*` to trigger it whenever the callback executes.

Reply:
Original intention was for the callback to handle the doubling of the shield via radiance.  Implemented this and pushed.